src/codec/framed_read.rs

```diff
@@ -110,7 +110,7 @@ fn decode_frame(
 
     if partial_inout.is_some() && head.kind() != Kind::Continuation {
         proto_err!(conn: "expected CONTINUATION, got {:?}", head.kind());
```

Comment:
is it possible to get this information into the error? `PROTOCOL_ERROR` is very generic and we lose details of  what specifically went wrong.

Reply:
PROTOCOL_ERROR is used a lot throughout RFC 7540. It sounds like you'd like internal errors to contain more detail than the protocol otherwise defines, e.g. by requiring two error types.

Reply:
A `GOAWAY` frame can include additional opaque debug data, which we can use to give a hint as to what the peer did. I think it'd be a good idea in most cases.

Reply:
Any additional information included in a GOAWAY frame is a potential security problem because the peer is untrusted and can probe things. Furthermore, data in the GOAWAY could be toxic, and discarded by the endpoints that receive it (see last para in https://datatracker.ietf.org/doc/html/rfc7540#section-6.8). Furthermore, some kinds of errors are specific to streams, and RST_STREAM can't carry any debug data.

So one school of thought is that most cases of error can be determined fairly straightforwardly by someone looking at the frames that were sent. A consistent error occurrence is usually strong enough grounds to cause someone to start taking a closer look at things. That's not to say making information available locally doesn't have merit, just don't rely on the GOAWAY debug data.

Reply:
Yea, all true. It's a balance, just like when returning 4xx errors from a server, you might include a message hinting at what the client did wrong, or you may decide you shouldn't. :shrug:

Reply:
One might be tempted to make the behaviour configurable, but that throws even more complication in :smile: